src/Coupon/WCCouponAdapter.php

```diff
@@ -75,9 +74,13 @@ public function mapTypes( $array ) {
 		$wc_coupon          = $array['wc_coupon'];
 		$this->wc_coupon_id = $wc_coupon->get_id();
 		$this->map_woocommerce_coupon( $wc_coupon );
+		if ( ! empty( $array['delete'] ) && $array['delete'] === true ) {
```

Comment:
There is a limitation here since we prefer to keep original start_date as it is and only update end_date when disable a promotion so Google keep an accurate record for this promotion. That's why we still need WC_Coupon as source when set effective date to expired. 

And from Google's perspective "set an effective date to expired" will work as disable the promotion as still possible to be enabled back later if e.g. a trashed promotion get recovered.

I can add comments there about that we need generate the promotion structures that we are used in disabling before coupon is completely deleted from the system.

Reply:
We could still have the object stored in a variable in the class that needs it (for example, in `SyncerHooks::handle_update_coupon`) and call the `disable_promotion` manually whenever we need to update the effective date. What I mean is that the additional `delete` flag that we're defining here for the `WCCouponAdapter` class can be removed because it's defining an external operation rather than an internal state or flag.

Reply:
Yes, just updated to  avoid add delete flag in adapter